Monika After Story/game/script-holidays.rpy

```diff
@@ -6903,16 +6903,34 @@ init -1 python:
         """
         checks if the given date is monikas birthday
         Comparison is done solely with month and day
-
         IN:
             _date - date to check. If not passed in, we use today.
         """
         if _date is None:
             _date = datetime.date.today()
 
+        _datetime = datetime.datetime.combine(_date, datetime.time())
+
+        return mas_isMonikaBirthday_dt(_datetime=_datetime)
+
+
+    def mas_isMonikaBirthday_dt(_datetime=None, _extend=0):
+        """
+        checks if the given date is monikas birthday.
+        Takes hours beyond the date into account via the `_extend` param.
+
+        IN:
+            _datetime - datetime to check. If not passed in, we use now.
+            _extend - int of how many hours we want to extend past 922
+                defaults to 0
+        """
+        if _datetime is None:
+            _datetime = datetime.datetime.now()
+
         return (
-            _date.month == mas_monika_birthday.month
-            and _date.day == mas_monika_birthday.day
+            _datetime.month == mas_monika_birthday.month
+            and _datetime.day == (mas_monika_birthday.day + int(_extend > 0))
+            and (_extend <= 0 or _datetime.hour < _extend)
         )
```

Comment:
I'm thinking of a bit more clear way, which would also allow us to use floats for hours (like `1.5` because just 1h is kind of short, or say half an hour for other cases).

```py
moni_bd_start = datetime.datetime.combine(mas_monika_birthday, datetime.time())
moni_bd_start = moni_bd_start.replace(year=datetime.date.today().year)

moni_bd_end = moni_bd_start + datetime.timedelta(days=1, hours=_extend)

return moni_bd_start <= _datetime < moni_bd_end
```

I'd also probably rename `_extend` into maybe `extend_by`.